src/scripting/lua_widget_attributes.cpp

```diff
@@ -514,10 +562,11 @@ int impl_widget_get(lua_State* L)
 {
 	gui2::widget& w = luaW_checkwidget(L, 1);
 	if(lua_isinteger(L, 2)) {
-
-		if(auto pwidget = find_child_by_index(w, luaL_checkinteger(L, 2))) {
-			luaW_pushwidget(L, *pwidget);
+		if(push_child_by_index(L, w, luaL_checkinteger(L, 2))) {
 			return 1;
+		} else {
+			// I don't know what to do here
+			DBG_LUA << "push_child_by_index() returned false";
```

Comment:
Maybe something like this?

```suggestion
			return luaL_argerror(L, 1, "index out of range");
```

Reply:
If all the possible failure cases are handled in push_child, then just remove the else. And if there's no path where push_child returns false, then you might as well remove the if and the return value too.